library/src/webapp/editor/ckextraplugins/common/js/itemsearch.js

```diff
@@ -91,28 +106,41 @@ $.fn.itemSearch = function(options) {
     var pagesContainer = $('<div/>').addClass('pages');
 
     // build page numbers and containers
-    for (i = 1; i <= pages; i++) {
-      nav.append($('<a/>').html(i).attr({'class': 'pageNum', 'href': '#', 'data-page': i}));
-      pagesContainer.append($('<div/>').attr({'class': 'page', 'data-page': i}));
-    };
 
+    var currentPage = $.fn.itemSearch.currentPage;
+    var iFrameId = settings.resultsContainer[0].id.replace('Results', 'Iframe');
+    if (currentPage >= 1 && currentPage  <=4){
+      if (currentPage  > 1){
+          appendBackButton(nav, currentPage, iFrameId );
+      }
+      for (i = 1; i <= 7; i++) {
+          appendPageButtons(i, currentPage , pagesContainer, nav, iFrameId);
+      }
+    }
+    else {
+        appendBackButton(nav, currentPage );
+        for (i = currentPage-3; i <= currentPage +3; i++) {
+            appendPageButtons(i, currentPage , pagesContainer, nav, iFrameId);
+        }
+    }
+
+    nav.append($('<a/>').html('<span class="pageSpan">Next »</span>').attr({'class': 'pageNum ' + iFrameId, 'href': '#', 'data-page': currentPage +1}));
     container.append(pagesContainer);
 
     // now move results into the correct containers
     for (i = 0; i < results.length; i++) {
-      var page = Math.ceil((i+1) / settings.pagination);
-      container.find('.page[data-page="' + page + '"]').append(results[i]);
+      container.find('.page[data-page="' + currentPage + '"]').append(results[i]);
     }
 
     $(document).on('click', '.pageNum', function() {
       // find closest results container
       var $this = $(this);
-      var localContainer = $this.closest('.paginated-results');
       var page = $this.data('page');
 
-      localContainer.find('.page').hide();
-      localContainer.find('.page[data-page="' + page + '"]').show();
-
+      $.fn.itemSearch.currentPage = page;
+      var searchResultsIFrameId = $.grep(this.classList, function(v) {return v.indexOf('Iframe')!=-1;});
+      var searchButtonLink = document.getElementById(searchResultsIFrameId).contentWindow.document.forms[0].children[1];
```

Comment:
I don't follow the code enough, but this looks live a very brittle way to get the search button as it's a very long change with multiple array indexes. What's the reason for this traversal?